java-frontend/src/test/java/org/sonar/java/resolve/SymbolTableTest.java

```diff
@@ -105,7 +105,8 @@ public void Generics() {
     JavaSymbol applyMethod = result.symbol("apply");
     assertThat(result.reference(83, 12)).isSameAs(applyMethod);
     assertThat(result.reference(89, 61)).isSameAs(applyMethod);
-    assertThat(applyMethod.usages()).hasSize(2);
+    // A method that does not compile is resolved whereas it should not.
```

Comment:
I would add an assertions with the reference line, to indicate which method call is wrongly resolved.